### DISCUSSION TITLE
pyb librairy with esp32

### DISCUSSION THREAD

**Discussion by spikey1973** [micropython/micropython#10627] [ESP32] [answered]
Probably a very basic question,

But can I use this PYB librairy also with esp32 boards (particulairy wemos lilo esp32 s2 mini)
I understand that the PYB librairy is particulail for the pyboards, unfortunately i find a lot of example codes (specifically stepper librairies) that use this pyb librairy. I looked at its functions, but it seems to be fairly extended, so this keeps me unsure and it remains unclear if I can use this librairy for my esp32 yes or no, and if not, if there is a replacement librairy one can use instead. 

any thoughts on the topic are appreciated. 

Kind greats 

Matthieu

**Comment by rkompass** [comment, marked as answer]
You can't use pyb library on other boards. In the long run pyb will be deprecated and it's functionality will be on the machine library. So if you use i2c or spi or pwm, it's better to start with machine, even on pyboard.
If you have a specific example that you are interested in that relies on pyb you could post it here, then we could try to help you change the code for machine. In principle that should be easy as long as the functionality is present on the esp32.
But consider that stm32f405 and esp32 are different architectures and they have different peripherals, so it might happen that you need a functionality that is present on the stm32f405 (pyboard) but not on the esp32 s2.
Also note that not all functionality that is there is available in the libraries. There is an ongoing endeavor to make machine universal, i.e. mostly hardware independent. This leads to a focus on a common set of functionalities and some more specific/rare peripherals might be neglected. That said I have to add that I did not program anything with stepper motors so far...

**Reply by spikey1973** [reply]
Thank you for this very informative and complete answer. I will Look around in the machine librairy and the pyb librairy and see which reference i should change to see if I get it working. if not I will post my enbdeavours here for help. Many thanks.
